src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -236,11 +250,35 @@ t8_forest_write_vtk_via_API (t8_forest_t forest, const char *fileprefix)
       }
       /* We insert the next cell in the cell array */
       cellArray->InsertNextCell (pvtkCell);
+      /*
+       * Write current cell Type in the cell Types array at the elem_id index.
+       * Depending on the values of the binary inputs write_treeid, 
+       * write_mpirank and write_element_id we also fill the corresponding
+       * arrays with the data we want(treeid,mpirank,element_id).
+       * To get the element id, we have to add the local id in the tree 
+       * plus theo
+       */
 
+      /* *INDENT-OFF* */
       cellTypes[elem_id] = t8_eclass_vtk_type[element_shape];
-      elem_id++;
+      if (write_treeid == 1) {
+        vtk_treeid->InsertNextValue (itree);
+      }
+      if (write_mpirank == 1) {
+        vtk_mpirank->InsertNextValue (forest->mpirank);
+      }
+      if (write_level == 1) {
+        vtk_level->InsertNextValue (scheme->t8_element_level (element));
+      }
+      if (write_element_id == 1) {
+        vtk_element_id->InsertNextValue (elem_id + offset + (long long)
+                                         t8_forest_get_first_local_element_id
+                                         (forest));
```

Comment:
Do we need the conversion to long long here? Why?

Reply:
It should work fine without the conversion. I removed it.

Reply:
So far, no update arrived here, may be you did not push?